Assets/Plugins/Clean/Editor/ComponentUpgrade.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using TMPro;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.UI;
+using Object = UnityEngine.Object;
+
+namespace Plugins.Clean.Editor
+{
+    public static class ComponentUpgrade
+    {
+        public static TextMeshProUGUI UpgradeText(Text t)
+        {
+            // Copy properties
+            var go = t.gameObject;
+            var properties = new ComponentProperties.Text(t);
+
+            // Swap Text component for TextMeshPropUGUI
+            Object.DestroyImmediate(t, true);
```

Comment:
FYI: in editor you should use DestroyImmediate, like it is used here. In playmode, it's better to use Destroy() instead as this will choose the time to actually do the destroy based on when the performance impact might be the least.

Reply:
👍